core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3188,6 +3201,28 @@ private void analyzeSelectAllColumns(
             }
         }
 
+        private List<Field> filterInaccessibleFields(List<Field> fields)
+        {
+            return fields.stream()
+                    .filter(field ->
+                                field.getOriginColumnName().isEmpty() ||
+                                field.getOriginTable().isEmpty() ||
+                                !filterInaccessibleColumns(
```

Comment:
Can we call this method once for all a collection of fields, not once per each field?